src/main/java/org/springframework/data/elasticsearch/core/AbstractDefaultIndexOperations.java

```diff
@@ -48,88 +53,158 @@
 	protected final ElasticsearchConverter elasticsearchConverter;
 	protected final RequestFactory requestFactory;
 
-	public AbstractDefaultIndexOperations(ElasticsearchConverter elasticsearchConverter) {
+	@Nullable protected final Class<?> boundClass;
+	@Nullable protected final IndexCoordinates boundIndex;
```

Comment:
`IndexCoordinates` should never be null. Rather, we should derive `IndexCoordinates` from the given class.